### PR TITLE
Improves ethereal APC sucking code

### PR DESCRIPTION
**Pull request by MacBlaze1**
## About The Pull Request

this pr fixes my shitcode into a less shitcode form. open to suggestions on how to improve intent mode code reusage.

## Why It's Good For The Game

responsibility is important

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
i dont think this needs a changelog, it has no player-facing changes

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -783,66 +783,63 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/auto_name, APC_PIXEL_OFFSET
 		return
 	if(!ishuman(user))
 		return
+	var/mob/living/carbon/human/apc_interactor = user
+	var/obj/item/organ/stomach/maybe_stomach = apc_interactor.getorganslot(ORGAN_SLOT_STOMACH)
+	if(!istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
+		togglelock(user)
```
Comment (Ryll-Ryll) [inline code/modules/power/apc.dm:789]:
```suggestion
	var/obj/item/organ/stomach/ethereal/maybe_stomach = apc_interactor.getorganslot(ORGAN_SLOT_STOMACH)
	if(!istype(maybe_ethereal_stomach))
		togglelock(user)
```
We only care about the stomach if it's an ethereal stomach type specifically, and `istype(thing, /path/we/want)` already returns false if `thing` is null, so we don't need the intermittent cast to a normal stomach

**Comment by Ryll-Ryll** [comment]
Just make that one change to see if it's an ethereal stomach directly and ping me and i'll merge it

**Comment by MacBlaze1** [comment]
> Just make that one change to see if it's an ethereal stomach directly and ping me and i'll merge it

@Ryll-Ryll did as you asked, had to update 1 other thing to get it to work

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -783,66 +783,63 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/power/apc/auto_name, APC_PIXEL_OFFSET
...
 		return
+	var/mob/living/carbon/human/apc_interactor = user
+	var/obj/item/organ/stomach/ethereal/maybe_ethereal_stomach = apc_interactor.getorganslot(ORGAN_SLOT_STOMACH)
+	if(!istype(maybe_ethereal_stomach))
+		togglelock(user)
 	else
-		var/mob/living/carbon/human/H = user
-		var/obj/item/organ/stomach/maybe_stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
-		if(!istype(maybe_stomach, /obj/item/organ/stomach/ethereal))
+		var/obj/item/organ/stomach/ethereal/stomach = maybe_ethereal_stomach
```
Comment (Ryll-Ryll) [inline code/modules/power/apc.dm:791]: this cast is redundant, you've already got maybe_ethereal_stomach that you can just use directly

**Comment by Ryll-Ryll** [comment]
one more thing

**Comment by MacBlaze1** [comment]
> one more thing

@Ryll-Ryll

**Pull request merged**